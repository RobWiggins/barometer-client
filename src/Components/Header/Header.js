import React from 'react';

export default class Header extends React.Component {
  
  render() {
    return (
      <header id="app-title" role="banner">
        <h1>Barometer</h1>
        <h2>See the chatter.</h2>
      </header>
    );
  }
}
