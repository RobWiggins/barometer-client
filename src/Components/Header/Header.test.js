import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('testing Header Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a Header component snapshot', () => {
    const wrapper = shallow(<Header />).find('#app-title');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});