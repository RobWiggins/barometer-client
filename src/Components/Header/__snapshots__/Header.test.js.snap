// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header Component renders a Header component snapshot 1`] = `
<header
  className="app-header"
  role="banner"
>
  <h1
    className="title-letter-holder"
  >
    <span
      className="app-title blue"
    >
      B
    </span>
    <span
      className="app-title blue-purple"
    >
      a
    </span>
    <span
      className="app-title purple"
    >
      r
    </span>
    <span
      className="app-title purple-pink"
    >
      o
    </span>
    <span
      className="app-title pink"
    >
      m
    </span>
    <span
      className="app-title red"
    >
      e
    </span>
    <span
      className="app-title red-orange"
    >
      t
    </span>
    <span
      className="app-title orange"
    >
      e
    </span>
    <span
      className="app-title orange-yellow"
    >
      r
    </span>
  </h1>
  <h2
    className="tagline"
  >
    See the chatter.
  </h2>
</header>
`;
